runtime/lua/tohtml.lua

```diff
@@ -1291,7 +1291,29 @@ local function opt_to_global_state(opt, title)
   if opt.font then
     fonts = type(opt.font) == 'string' and { opt.font } or opt.font --[[@as (string[])]]
   elseif vim.o.guifont:match('^[^:]+') then
-    table.insert(fonts, vim.o.guifont:match('^[^:]+'))
+    -- Example:
+    -- Input: "Font,Escape\,comma, Ignore space after comma"
+    -- Output: { "Font","Escape,comma","Ignore space after comma" }
+    local str, char_pre = '', ''
+    for _, char in ipairs(vim.split(vim.o.guifont:match('^[^:]+'), '')) do
+      if char == ',' then
+        if char_pre == '\\' then
+          str = str .. char
+        else
+          table.insert(fonts, str)
+          str = ''
+        end
+      elseif char ~= '\\' and not (char == ' ' and char_pre == ',') then
+        str = str .. char
+      end
+      char_pre = char
+    end
+    table.insert(fonts, str)
+  end
+  -- Generic family names (monospace here) must not be quoted
+  -- because the browser recognizes them as font families.
+  for i, v in pairs(fonts) do
+    fonts[i] = ('"%s"'):format(v)
```

Comment:
not needed if you use the code from https://github.com/neovim/neovim/pull/30105/files#r1741902695